documentation/Glossary.md

```diff
@@ -0,0 +1,149 @@
+
+## DataCooker
+
+Component of a [Processing Pipeline](#processing-pipeline) that consumes data from one or more sources and optionally outputs new data for other components to consume. Conceptually, a DataCooker "cooks" data from one type to another.
+
+See also: [SourceDataCooker](#sourcedatacooker) and [CompositeDataCooker](#compositedatacooker).
+
+## DataSource
+
+An input to be processed by a [CustomDataProcessor](#customdataprocessor). In most situations, this is a `FileDataSource` that contains the full path to a file specified by a user. 
+
+## DynamicTable
+
+A [Table](#table) that is built outside of the standard [table building cycle](#table-building-cycle). DynamicTables are commonly built in response to a [TableCommand](#tablecommand) being invoked.
+
+## Plugin
+
+A collection of one-or-more [ProcessingSources](#processingsource) that collectively create zero-or-more [Tables](#table). Plugins also consist of zero-or-more [DataCookers](#datacooker).
+
+## Processing Pipeline
+
+A collection of [DataCookers](#datacooker) and [SourceParsers](#sourceparser) that define a structured flow of data.
+
+## ProcessingSource
+
+An entrypoint for a [Plugin](#plugin). A ProcessingSource
+1. Declares a human-readable name and description
+2. Defines the [DataSource(s)](#datasource) it supports
+3. Creates the [CustomDataProcessor](#customdataprocessor) that will process instances of supported DataSources
+4. Advertises the [Table(s)](#table) that may be built in response to DataSources being processed
+
+## Projection
+
+A function that maps a row index for a [Table](#table) to a piece of data. Conceptually, a Projection is combined with a [ColumnConfiguration](#columnconfiguration) to complete define a Table's column: the ColumnConfiguration defines the name and metadata about a column, and its associated Projection defines the column's data.
+
+## Simple Table
+
+A [Table](#table) variant that cannot participate in a [Processing Pipeline](#processing-pipeline) and must be built by a [CustomDataProcessor](#customdataprocessor). Simple Tables should only be used if your plugin does not use [DataCookers](#datacooker). However, since in most data-processing situations it is recommended to use DataCookers to architect your data-processing, it is recommended to use only standard [Tables](#table).
+
+## SourceDataCooker
+
+A [DataCooker](#datacooker) that receives input from a specific [SourceParser](#sourceparser). A SourceDataCooker may also receive input from other DataCookers.
+
+## SourceParser
+
+An object that a [CustomDataProcessor](#customdataprocessor) typically tasks with parsing a [DataSource](#datasource) into individual records. The records that a SourceParser emits typically begin a [Processing Pipeline](#processing-pipeline), wherein [DataCookers](#datacooker) further process data to be consumed by [Tables](#table).  
```

Comment:
>typically [](http://example.com/codeflow?start=61&length=9)

do we need the word 'typically' here?

Reply:
In theory if no cookers are hooked up to the parser nothing would happen, right?